app/src/main/java/org/mozilla/tv/firefox/utils/PeriodicRequester.kt

```diff
@@ -0,0 +1,91 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.utils
+
+import android.support.annotation.VisibleForTesting
+import io.reactivex.Maybe
+import io.reactivex.Observable
+import io.reactivex.Single
+import java.util.concurrent.TimeUnit
+import kotlin.math.pow
+
+private val CACHE_UPDATE_FREQUENCY_MINUTES = 45L
+private val CACHE_UPDATE_FREQUENCY_SECONDS = TimeUnit.MINUTES.toSeconds(CACHE_UPDATE_FREQUENCY_MINUTES)
+
+/**
+ * Simple interface for making a network request.
+ *
+ * Failed requests should be represented as [Maybe.empty].
+ *
+ * Should operate on a background thread.
+ */
+interface Endpoint<T> {
+    fun request(): Single<Response<T>>
+}
+
+sealed class Response<T> {
+    data class Success<T>(val data: T) : Response<T>()
```

Comment:
is there a reason you wouldn't use the `onSuccess` and `onError` cases of a `Single` instead of using this custom `Response` type?

Reply:
Oh, I see, you want to use this in an `Observable` sometimes....

Reply:
Yeah, and the repo also needs to be aware of failure responses for some of its logic.  The observable between the requester and the repo needs to stay open even after failure responses, so we can't use `onError` there.

Reply:
So this would be more on the data abstraction layer? Data transformation feed?